src/adera/Machines/UserControl.cpp

```diff
@@ -129,42 +107,79 @@ void SysMachineUserControl::update_sensor(ActiveScene &rScene)
     for (ActiveEnt ent : view)
     {
         MachineUserControl &machine = view.get<MachineUserControl>(ent);
-        auto& throttlePos = std::get<wiretype::Percent>(machine.m_woThrottle.value()).m_value;
 
-        float throttleRate = 0.5f;
-        auto delta = throttleRate * rScene.get_time_delta_fixed();
+        // Get the Percent Panel which contains the Throttle Port
+        auto *panelPercent = rScene.reg_try_get< ACompWirePanel<Percent> >(ent);
 
-        if (!machine.m_enable)
+        if (panelPercent != nullptr)
         {
-            continue;
+            WirePort<Percent> const *portThrottle
```

Comment:
I remember consts not needing the Hungarian prefixes; I guess that's only for references?

The reasoning for using "port for throttle" instead of "throttle port" is that I usually put the common word within a sequence of similar variables first to improve readability. In this case there's `portThrottle` and `portAttitude`. It would be harder to tell the variables are similar if they were names `throttlePort` and `attitudePort`.

Reply:
The reason for the p is to indicate its a pointer. Local constant variables don't need "c_" or anything.